entity.go

```diff
@@ -84,9 +84,9 @@ func NewMultipart(header Header, parts []*Entity) (*Entity, error) {
 // If the message uses an unknown transfer encoding or charset, Read returns an
 // error that verifies IsUnknownCharset, but also returns an Entity that can
 // be read.
-func Read(r io.Reader) (*Entity, error) {
+func Read(r io.Reader, opts *textproto.ReadOpts) (*Entity, error) {
```

Comment:
In the future, we'll probably want to have `message.ReadOptions` to specify a custom charset reader for instance. I wonder if we should introduce it now to avoid breaking the API twice.

Also, should we have `message.ReadWithOptions` instead of breaking the API?

Reply:
>Also, should we have message.ReadWithOptions instead of breaking the API?

Since go-message is using Go modules and is 0.X, I do not care much. Library users are expected to pin a specific library version. However, I think there might be a valid reason: most users probably do not care about options at all so having simply Read without options can be useful.

Reply:
I agree that most users do not care about options. Also is cleaner to have no options instead of writing `nil` while using a generic `Read` function. Can we use functional options. For example:

```
package message
...
...
type ReadOptions struct {
	Header textproto.ReadOptions
}

type ReadOption func(*ReadOptions)
...
...
func Read(r io.Reader, options ...ReadOption) (*Entity, error) {
	readOption := &ReadOptions{}
	for _, option := range options {
		option(readOption)
	}

	br := bufio.NewReader(r)
	h, err := textproto.ReadHeader(br, &readOption.Header)
```

Then have our custom functions
```
func SetMaxHeaderLineLength(n int) ReadOption {
	return func(options *ReadOptions) {
		options.Header.MaxHeaderLineLength = n
	}
}
```

Reply:
Let's keep it simple and just have a basic options struct.